Source/Moya.swift

```diff
@@ -65,12 +87,16 @@ public class MoyaProvider<Target: TargetType> {
     }
     
     /// Returns an Endpoint based on the token, method, and parameters by invoking the endpointsClosure.
-    public func endpoint(token: Target) -> Endpoint<Target> {
+    public func endpoint(token: TargetType) -> Endpoint<TargetType> {
         return endpointClosure(token)
     }
     
     /// Designated request-making method. Returns a Cancellable token to cancel the request later.
     public func request(target: Target, completion: Moya.Completion) -> Cancellable {
+        guard let target = target.resource else {
```

Comment:
What Ash describes here is also what I was imagining - start with what we have, instead of adding anything new, just update the examples to show that *struct based endpoints is already possible*.